app/code/Magento/Catalog/Model/Product/Image/Cache.php

```diff
@@ -127,4 +154,93 @@ protected function processImageData(Product $product, array $imageData, $file)
 
         return $this;
     }
+
+    /**
+     * Get themes in use
+     *
+     * @return array
+     */
+    public function getThemesInUse()
+    {
+        $themesInUse = [];
+
+        $registeredThemes = $this->themeCollection->loadRegisteredThemes();
+        $storesByThemes   = $this->themeCustomizationConfig->getStoresByThemes();
+
+        foreach ($registeredThemes as $registeredTheme) {
+            if (array_key_exists($registeredTheme->getThemeId(), $storesByThemes)) {
+                $themesInUse[] = $registeredTheme;
+            }
+        }
+
+        $connection = $this->attribute->getResource()->getConnection();
```

Comment:
Getting dependencies in such way (one dependency via another one) violates the principle of least knowledge. 
Please consider refactoring this.